src/components/coin.ts

```diff
@@ -122,17 +134,19 @@ export const changeDbCoinBalanceByUserId = async (
   reason: string | null,
   adminId: string | null
 ): Promise<void> => {
-  const db = await openDB();
-  await db.run(
-    `
-    INSERT INTO user_coin (user_id, balance) VALUES (?, ?)
-    ON CONFLICT(user_id)
-    DO UPDATE SET balance = ?`,
-    userId,
-    newBalance,
-    newBalance
-  );
-  await createCoinLedgerEntry(userId, oldBalance, newBalance, event, reason, adminId);
+  try {
+    const db = await openDB();
+    await db.run(
+      `
+      INSERT INTO user_coin (user_id, balance) VALUES (?, ?)
+      ON CONFLICT(user_id)
+      DO UPDATE SET balance = ?`,
+      userId,
+      newBalance,
+      newBalance
+    );
+    await createCoinLedgerEntry(userId, oldBalance, newBalance, event, reason, adminId);
+  } catch (e) {}
```

Comment:
> In addition, in coin.ts, there appears to be error catching, in which the error is then logged or nothing is done at all. In these cases, do we also want the user to be aware that there is an error?

This is one of the few instances of what I meant by "nothing is done at all."

Reply:
Similar to your previous comment - I've fixed this